test/metabase/test/util/dynamic_redefs.clj

```diff
@@ -17,6 +17,9 @@
 (defn- var->proxy
   "Build a proxy function to intercept the given var. The proxy checks the current scope for what to call."
   [a-var]
+  (assert (ifn? @a-var) "Cannot proxy non-functions")
+  (assert (not (keyword? @a-var)) "Cannot proxy keywords")
+  (assert (not (coll? @a-var)) "Cannot proxy collections")
```

Comment:
A bit torn here about allowing sets - those are often just shorthand for functions.

Reply:
Thought about this more, and there were only two cases in our existing tests where the original definition was a set, and overriding them was indeed OK. 

Even in those cases, there's a chance we'd change code in future to depend on them being actual collections, and breaking tests in really confusing and even flaky ways.

I think being strict like this is worth any potential awkwardness where we force original definitions to be explicit or wrap a set.